thumbor/engines/extensions/exif_orientation_editor.py

```diff
@@ -0,0 +1,66 @@
+import os
+import struct
+from io import BytesIO
+
+PREFIXES = [
+    b"MM\x00\x2A",  # Valid TIFF header with big-endian byte order
+    b"II\x2A\x00",  # Valid TIFF header with little-endian byte order
+    b"MM\x2A\x00",  # Invalid TIFF header, assume big-endian
+    b"II\x00\x2A",  # Invalid TIFF header, assume little-endian
+]
+
+EXIF_HEADER = b"Exif\x00\x00"
+
+
+class ExifOrientationEditor():
+    _endian = "<"
+    _offset = None
+
+    def __init__(self, data):
+        if data[:6] != EXIF_HEADER:
+            raise SyntaxError("not a TIFF file (header %r not valid)" % data[:4])
+
+        self.fp = BytesIO(data[6:])
```

Comment:
What is `fp`?

Reply:
File pointer. 
Struggling to find better name. Any suggestions?

Reply:
Well, since ... I think ... what you are doing here is declaring a _named_, binary buffer for `data[6:]` range of bytes ... we might be able to call this `exif_buffer`?

Reply:
I agree with changing the names to more descriptive variables. Great work btw.